app/src/main/java/fr/groggy/racecontrol/tv/f1tv/F1TvClient.kt

```diff
@@ -117,12 +151,60 @@ class F1TvClient @Inject constructor(
                 )
             }
         } catch (e: Exception) {
-            /*
-             * The pre seasons for example are not available to query
-             * on the regular api, for this fallback to the archive
-             */
             Log.d(TAG, "getF1TvSessions failed with ${e.message}")
-            //return getSessionArchive(event, season)
+            return listOf()
+        }
+    }
+
+    private suspend fun getFutureF1TvSessions(event: F1TvSeasonEvent): List<F1TvSession> {
+        val dateTimeFormatter: DateTimeFormatter =
+            DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm")
+                .withZone(ZoneId.systemDefault())
+        try {
+            val response = get(
+                LIST_FUTURE_SESSIONS.format(getCurrentLocale(), event.meetingKey),
+                futureSessionResponseJsonAdapter
+            )
+            Log.d(TAG, "Fetched future sessions for event ${event.id}")
+            val schedules = ArrayList<F1TvFutureSessionEvent>()
+            response.resultObj.containers
+                .filter { it.layout == "schedule" }
+                .forEach { it ->
+                    it.retrieveItems.resultObj.containers
+                        .filter { it.eventName.equals("ALL") }
+                        .forEach { ev ->
+                            ev.events!!
+                                .filter {
+                                    it.metadata.emfAttributes.sessionStartDate > Instant.now()
+                                        .toEpochMilli()
+                                }
+                                .forEach { fev ->
+                                    run {
+                                        schedules.add(fev)
+                                        Log.d(TAG, fev.toString())
+                                    }
+                                }
+                        }
+                }
+            return schedules.map {
+                F1TvSession(
+                    id = F1TvSessionId(it.id),
+                    eventId = event.id,
+                    pictureUrl = PICTURE_URL.format(it.metadata.pictureUrl),
+                    contentId = it.metadata.contentId,
+                    name = it.metadata.title,
+                    contentSubtype = dateTimeFormatter.format(Instant.ofEpochMilli(it.metadata.emfAttributes.sessionStartDate)),
```

Comment:
I'm abusing the `contentSubType` to allow showing the start time of the future session. What is the better way to do this?

Reply:
Well, if we had something better in place I'd say to change, but I think for now at least is ok.

Reply:
Great.